### ISSUE TITLE
Automatic Cover Download

### ISSUE THREAD

**Issue by padremayi** [DNA64/SuperHakchi#5]
Hi,
I don't find a way to contact DNA64 and I write here ;-)

If you want we have an app that automatically download SNES cover
starting from a roms folder and create, for each rom, a folder named
as rom filename. Inside this folder there is another folder progressively named
CLV-X-XXXXX with CLV-X-XXXXX.png and CLV-X-XXXXX_small.png files (with
correct resolution).

In these way the procedure is automatic and very fast.

If you are interested, let me know ;-)

Regards

**Comment by DNA64** [comment]
Hi,

That sounds great. Where does it get the artwork from, and what are the image sizes?
(Hakchi2 has these features but the images aren't the best, and the image dimensions are wrong (not matching whats on the system, not that those are perfect either)  as of hakchi2_0.20.rc4)

**Comment by padremayi** [comment]
"Where" is a secret because is a part of a closed source code from another program ;-)

Cover CLV-X-XXXXX.png is 228x160 while CLV-X-XXXXX_small.png is 40x28 as you wrote in your guide.

At the moment it works on Ubuntu, I'm testing it for Windows because your guide use Windows.

**Comment by DNA64** [comment]
Hakchi2 has now been officially released though I have a few issues with it and suggestions I need to pass on to ClusterM <3  Though I gather my ideas are already on the drawing board and it's just a matter of time. 

Basically I don't like the way it googles for the images, and we can't edit the google search string like we use to be able to do with the NESCE. 

It would also be nice to have something like www.wiitdb.com for the NES/SNES Classic.

**Comment by padremayi** [comment]
Ok, the app is ready to be used on Windows, you will find it at the link in the first post of this thread:
http://www.retropie-italia.it/viewtopic.php?f=24&t=585

All you have to do is to put the "roms" folder in the same folder where SNES Mini Cover Downloader v1.0 is present, than launch it!

Supported file extension: .bin, .smc, .sfc, .fig, .swc, .mgd, .zip

I tested it on Windows 10 x64, let me know if it correctly works on different version.

Don't forget to cite our work in your guide ;-)

**Comment by DNA64** [comment]
Excellent, I'll check it out. And I will certainly give you credit in the guide :)

**Comment by padremayi** [comment]
Did you try the app? If not I've just compiled the 1.1 version with a little adjustment in a message in case of database error. I use a 32 bit compiler in order to guarantee the compatibility with both 32 and 64 bit system

**Comment by DNA64** [comment]
Did you release 1.1? I'm updating the guide.

**Comment by padremayi** [comment]
Yes for both 32/64 bit architecture and a little changes on a debug message. You'll find it at the same link above

**Comment by DNA64** [comment]
I finally had a chance to test it out and as impressive as it was, unfortunately it was unable to find several images. It also used several of the wrong images. 

Hakchi is much more effective and uses http://thegamesdb.net/ along with CRC32 Checksums to correctly identify the correct boxart. You may want to see about implementing similar features into your script for pi users.

**Comment by padremayi** [comment]
Yes, many covers are missing because are present the in-game snapshot, it is possibile to add the feauture to insert this one when cover isn't present.

I already add the feature to ask to the user what to do with 4 options:
1. Save always
2. Skip when accuracy is lower than a treshold inserted by the user
3. Ask what to do when accuracy is lower than the inserted treshold
4. Ask always

I'm also adding CRC32 features ;-)
